platforms/android/src/org/crosswalk/engine/XWalkCordovaUiClient.java

```diff
@@ -172,12 +175,15 @@ public void onPageLoadStopped(XWalkView view, String url, LoadStatus status) {
     // File Chooser
     @Override
     public void openFileChooser(XWalkView view, final ValueCallback<Uri> uploadFile, String acceptType, String capture) {
-        uploadFile.onReceiveValue(null);
+        if (mFileChooser == null) {
+            mFileChooser = new XWalkFileChooser(parentEngine.cordova.getActivity());
```

Comment:
Now XWalkView can use any context. The developer could pass an application context into it and hold it in an activity. So casting the context inside of XWalkView to activity is not a proper option.